### ISSUE TITLE
no-undef rule can not work in latest eslint version

### ISSUE THREAD

**Issue by yinhaibo01** [eslint/eslint#6389]
**What version are you using?**
2.4.0
**What did you do?**
run eslint with recommand rule.
**What happened?**
Can not check the no-undef error.
**What did you expect to happen?**
I have install the eslint 1.10.3 in another PC, the no-undef error will throw.
But in latest eslint, this error has been ignored.
I expect this error should be checked in latest version.

the test javascript content is this:

```js
var arr = [0, 1, 2, 3, 4];
length = arr.length;
```

the config file is this:
```js
{
  "env": {
    "browser": true,
    "node": true
  },
  "settings": {
  },
  "extends": "eslint:recommended",
}
```

check result is different between eslint@2.11.1 and eslint@1.10.3

**Comment by eslintbot** [comment]
Hi @yinhaibo01, thanks for the issue. It looks like there's not enough information for us to know how to help you. If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by alberto** [comment]
I can't reproduce. 

Latest version is 2.12.0, but you mention 2.4.0 and 2.11.1. Which version is it?

**Comment by eslintbot** [comment]
Hi @yinhaibo01, thanks for the issue. It looks like there's not enough information for us to know how to help you. If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by ilyavolodin** [comment]
@alberto 2.11.1
I also can't reproduce. I get:
```js
2:1 - 'length' is not defined. (no-undef)
```

**Comment by mysticatea** [comment]
`browser` environment has `length` global variables, so `no-undef` does not warn it. `length` is defined.

**Comment by mysticatea** [comment]
I think #5358 would solve this issue.

**Comment by alberto** [comment]
Oh, I missed that part. Thanks @mysticatea. Closing this then, since #5358 also exists.

**Comment by yinhaibo01** [comment]
Thank you for your reply, I will update the latest version.

**Comment by mysticatea** [comment]
@yinhaibo01 #5358 is still opened. It would be solved in the next version (maybe 2.13.0), I hope. :wink:
